haystack/nodes/retriever/_embedding_encoder.py

```diff
@@ -105,21 +105,6 @@ def __init__(self, retriever: "EmbeddingRetriever"):
             num_processes=0,
             use_auth_token=retriever.use_auth_token,
         )
-        # Check that document_store has the right similarity function
```

Comment:
This seems unrelated. Is it being checked elsewhere now?

Reply:
Yes: https://github.com/deepset-ai/haystack/pull/3455 I didn't want to make this check compatible with the new signature given that that other PR was already changing this code path